core/domain/learner_progress_services.py

```diff
@@ -0,0 +1,231 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for tracking the progress of the learner."""
+
+from core.platform import models
+from core.domain import subscription_services
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+def add_exp_id_to_completed_list(user_id, exp_id):
+    """Adds the exploration id to the completed list of the user.
```

Comment:
I think this (and other) functions need to specify the contract more clearly: Adds the exploration id to ..., unless ....

Also, docstrings should be in standard format (with Args, Returns, etc.)